models/path.py

```diff
@@ -122,6 +125,14 @@ def set_controller(cls, controller=None):
         cls.controller = controller
 
     @staticmethod
+    @retry(
+        stop=stop_after_attempt(3),
+        wait=wait_combine(wait_fixed(3), wait_random(min=0, max=5)),
+        retry=retry_if_result(lambda result: result == []),
```

Comment:
This makes sense considered how the `get_paths` return was structured, but notice that it's still lacking to handle request errors, like timeout won't be handled correctly, `log.error` would be logged on each retry (during a retry we have log.warning with tenacity already), so instead of the log.error below in the orignal func, it should reraise a retry-able request exc. Since you're at it, let's go ahead and improve this part too. Something similar that you've implemented on of_lldp https://github.com/kytos-ng/of_lldp/blob/master/main.py#L255-L271

Reply:
Refactored with new exception `PathFinderException`